### PR TITLE
Project Status Page (5.4 branch) 

### PR DESCRIPTION
**Pull request by dwelch2344**
With Protractor becoming LTS and given the partnership with [HeroDevs](https://www.herodevs.com/) to provide support past August 2023, we had discussed adding documentation to explain what the future holds.

The bulk of this is adding a `banner` to every page and a `status page` that gives more information about EOL. Both of these have 3 minor variants of verbiage, depending on the browser's date on render. Screenshots below:

## Now - Aug 2023
current

## Sept 2023 - Aug 2024
sept 2023 to aug 2024

## Sept 2024+
sept 2024

_Note_: This PR replaces #5557 (and #5556) to both address EOL and update the docs / builds for the website. Apologies for another PR, but it looks like the current site is currently running off a 5.4 commit that never made it to the release branch.

### CODE REVIEW

File: lib/selenium-webdriver/webdriver.js [hunk 1]
```diff
@@ -236,7 +236,7 @@ webdriver.WebDriver.prototype.executeAsyncScript = (script, var_args) => {};
  * button.click();
  * browser.call(logText, counter);
  *
- * @param {function(...): (T|promise.Promise<T>)} fn The function to
+ * @param {function(...*): (T|promise.Promise<T>)} fn The function to
```
Comment (mgechev) [inline lib/selenium-webdriver/webdriver.js]: If this change is not related to the banner update, I'd recommend reverting back. I assume you're not able to build without it?

File: website/README.md [hunk 2]
```diff
@@ -6,21 +6,39 @@ Generates the documentation for the protractor website.
 ## How to run
 
 ```shell
-npm run compile_to_es5
+npm i
+npx gulp
 cd website/
 npm install
-npm run build
+npx gulp
```
Comment (dgp1130) [inline website/README.md]: Can you include the version of `gulp` you're using for future reference? That way updates to `gulp` don't break this process.

File: website/README.md [hunk 3]
```diff
@@ -6,21 +6,39 @@ Generates the documentation for the protractor website.
...
+#
+# Fatal error in , line 0
+# Check failed: !value_obj->IsJSReceiver() || value_obj->IsTemplateInfo().
+#
+#
+#
+#FailureMessage Object: 0x306a626c0
+```
+
+Eventually the project should migrate to Gulp 5 / Node 18+ to solve this issue, but in the meantime a simple workaround to run `npm run arm64_repair` (after every `npm install` in the `website/` directory)
```
Comment (dgp1130) [inline website/README.md]: Should this be included in the "How to run" instructions above?

File: website/partials/status.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    <p>
+      Although active development on Protractor ended in December of 2022, the project continued to be supported under <strong>Long Term Support</strong> with critical security &amp; compatibility fixes via the community through August 2023.
+    </p>
+    <p>
+      As of September 2023, Protractor's status is officially <strong>End Of Life</strong>.
+    </p>
+  </div>
+  
+  <h2>What does End Of Life mean?</h2>
+  The packages will remain accessible on GitHub, npm, Bower, and Release archive. This website will also remain here indefinitely. The project will be archived on GitHub and remain accessible indefinitely, but will no longer be open to issues / comments / etc.
```
Comment (dgp1130) [inline website/partials/status.html]: > This website will also remain here indefinitely.

@mgechev, are we comfortable committing to the website staying up _forever_? I doubt GitHub Pages would push a significant breaking change here, but if they did and it was a non-trivial effort to move off it, I expect we probably wouldn't bother given Internet Archive as the ultimate workaround.

**Review by dgp1130 — CHANGES_REQUESTED** [review]
I recommend squashing these commits, as the last two don't provide a lot of value and just fixup stuff from the first.

Wording of the messages sounds good to me, just one note about the website lifetime.

File: website/partials/status.html (hunk 4 above)
Comment (mgechev) [reply website/partials/status.html]: Yeah, I agree. Let's change to "foreseeable future"

We're not anticipating making the website unavailable, but inevitably it'll happen one day.

File: lib/selenium-webdriver/webdriver.js (hunk 1 above)
Comment (dwelch2344) [reply lib/selenium-webdriver/webdriver.js]: Happy to do so! Yeah, I'm a little concerned about the build process to actually deploy this as there were a few docs / build scripts that felt like they may be half-updated. I was able to get it to build, but it required updating the `dgeni` version for a bug fix (and thus where this came in)

File: website/README.md (hunk 2 above)
Comment (dwelch2344) [reply website/README.md]: I believe `npx gulp` will first look in the `node_modules/.bin/` first, so it should pin directly to whatever's listed in the `package.json`. That said, happy to include the specific version if preferable!

**Comment by dwelch2344** [comment]
@mgechev @dgp1130 Thanks for the review! Cleaned up 3 of the 4 comments, left a question on the final one for @dgp1130

File: website/README.md (hunk 2 above)
Comment (dgp1130) [reply website/README.md]: Oh I didn't realize that was properly listed as a dev dependency. How about we make `"gulp": "gulp"` a script and then document it as `npm run gulp`. That way it's specifically tied to the installed dependency, not downloaded over the network (ex. if the user forgot to `npm install` first).

**Review by mgechev — APPROVED** [review]

File: website/README.md (hunk 2 above)
Comment (dwelch2344) [reply website/README.md]: Great call! I had totally missed it, but `npm run build` would essentially do just that

**Comment by dwelch2344** [comment]
@dgp1130 I think we're good to go. I squashed to a single commit and cleaned up the issues, but definitely let me know if  there's anything else we can do on our end! Thanks

**Review by dgp1130 — APPROVED** [review]
Yep, looks great. Thanks @dwelch2344!

**Comment by dgp1130** [comment]
I'll take a look at building and pushing to `gh-pages` tomorrow to see if we can get this out.

**Comment by dgp1130** [comment]
Sorry, got distracted with v16. Pushed the release now and protractortest.org seems to be updated correctly.

I needed to add back in that typing change to get it to build, not sure how you were avoiding that. I also needed to swap a boolean condition as it seems the header message was displayed in the wrong order.

https://github.com/angular/protractor/commit/bb0e475b449fe2ed2a71c2e63bc1ab46f7a03309
